ethcontract/src/contract/method.rs

```diff
@@ -196,6 +196,21 @@ impl<T: Transport, R: Tokenize> ViewMethodBuilder<T, R> {
         self
     }
 
+    /// Specify the nonce for the transation, if not specified will use the
+    /// current transaction count for the signing account.
+    pub fn nonce(mut self, value: U256) -> Self {
+        self.m = self.m.nonce(value);
+        self
+    }
+
+    /// Specify the number of confirmations to wait for when confirming the
+    /// transaction, if not specified will wait for the transaction to be mined
+    /// without any extra confirmations.
+    pub fn confirmations(mut self, value: usize) -> Self {
+        self.m = self.m.confirmations(value);
+        self
+    }
```

Comment:
I just remembered, this doesn't do anything for view methods AFAIK (because they don't get mined so it doesn't make sense to wait for block confirmations).

I agree with the `nonce()` helper method though!

Reply:
Because the method returns a state that might be changed in the latest blocks, and when reorg happened the state will be reverted, which makes different views of a data for each call. I think confirmations() would prevent that.

Reply:
But the `ViewMethodBuilder` doesn't make use of `confirmations` in the `call()`.

I think in order to achieve that at the moment, you would need to specify the `block_number` for calling (and not the `confirmations`).

Reply:
I see thanks for the clarification, fixed the PR.